kpatch-build/create-diff-object.c

```diff
@@ -228,6 +230,24 @@ static struct rela *toc_rela(const struct rela *rela)
 				   (unsigned int)rela->addend);
 }
 
+/*
+ * Mapping symbols are used to mark and label the transitions between code and
+ * data in elf files. They begin with a "$" dollar symbol. Don't correlate them
+ * as they often all have the same name either "$x" to mark the start of code
+ * or "$d" to mark the start of data.
+ */
+static bool kpatch_is_mapping_symbol(struct kpatch_elf *kelf, struct symbol *sym)
```

Comment:
I think we can change the function signature to `static bool kpatch_is_mapping_symbol(enum architecture arch, struct symbol *sym)`. 
We only need the arch information, not the whole elf here.

There are only 2 usage for this function, `kpatch_replace_sections_syms` and `kpatch_correlate_symbols`.
There would be no difference for the `kpatch_replace_sections_syms`, as it is already a very complex function.

But for the `kpatch_correlate_symbols`, I feel it should have a similar signature as the `kpatch_correlate_sections`. But in the current code, it takes two `kpatch_elf*` as input parameters, only for getting this arch information. 

I know we have to have some compromise due to problem complexity. But I think, comparing to have two `kpatch_elf*`,  `kpatch_correlate_symbols(struct symbol *sym_orig, struct symbol *sym_patched, enum architecture arch)` looks more reasonable to me.

@swine, If you are fine with this change, I can create another PR to your branch later.